subprojects/core/src/main/java/org/gradle/initialization/layout/ProjectCacheDir.java

```diff
@@ -43,8 +48,20 @@ public File getDir() {
         return dir;
     }
 
+    public void delete() {
+        deleteOnStop = true;
+    }
+
     @Override
     public void stop() {
+        if (deleteOnStop) {
+            try {
+                deleter.deleteRecursively(dir);
+            } catch (IOException e) {
+                LOGGER.debug("Failed to delete unused project cache dir " + dir.getAbsolutePath(), e);
+            }
+            return;
+        }
```

Comment:
Shouldn't we handle this as part of `VersionSpecificCacheCleanupAction`? If the user accidentally deletes the settings file and runs a task, will he lose all the caches?

Reply:
The idea here is that in the end there will be no ".gradle" folder left somewhere on your disc where you accidentally started Gradle. 

Another option would to just not care. But I also think it is not such a big deal to "loose" the history of the project cache.   The most valuable caches are in the user home anyway. And I am not sure that a scenario where the user deletes the file by mistake and then runs Gradle is very common.

Reply:
👍